app/models/katello/sync_plan.rb

```diff
@@ -16,7 +16,7 @@ class SyncPlan < Katello::Model
 
     belongs_to :organization, :inverse_of => :sync_plans
     has_many :products, :class_name => "Katello::Product", :dependent => :nullify
-
+    belongs_to :foreman_tasks_recurring_logic, :inverse_of => :sync_plans
```

Comment:
I think here, you need to specify the class name: :class_name => "::ForemanTasks::RecurringLogic"